File: Java_Programming/Practise_Programs/PP_030.java
//////////////////////////////////////////////////////////////////
//
//  File Name         : PP_030
//
//  Description       : This program accepts a number from the user
//                      and counts the number of even digits present
//                      in the given number.
//
//  Author            : Shubham Somanath Gadhe
//  Date              : 10/07/2026
//
//  Time Complexity   : O(n)
//  Space Complexity  : O(1)
//
//////////////////////////////////////////////////////////////////

import java.util.Scanner;

class DigitX
{
    //////////////////////////////////////////////////////////////////
    //
    //  Method Name      : CountEvenDigits()
    //
    //  Description      : It is used to count the number of even
    //                     digits present in the given number by
    //                     extracting each digit and checking
    //                     whether it is even.
    //
    //  Author           : Shubham Somanath Gadhe
    //  Date             : 10/07/2026
    //
    //////////////////////////////////////////////////////////////////

    public int CountEvenDigits(int iNo)
    {
        int iDigit = 0;
        int iCount = 0;

        while(iNo != 0)
        {
            iDigit = iNo % 10;

            if(iDigit % 2 == 0)
            {
                iCount++;
            }

            iNo = iNo / 10;
        }

        return iCount;
    }
}

class PP_030
{
    public static void main(String A[])
    {
        Scanner sobj = new Scanner(System.in);
        DigitX dobj = new DigitX();

        int iValue = 0;
        int iRet = 0;

        System.out.println("Enter Number : ");
        iValue = sobj.nextInt();

        iRet = dobj.CountEvenDigits(iValue);

        System.out.println("Number of Even Digits are : " + iRet);
    }
}

//////////////////////////////////////////////////////////////////
//
//  Sample Input  :
//      Enter Number :
//      123456
//
//  Sample Output :
//      Number of Even Digits are : 3
//
//////////////////////////////////////////////////////////////////

File: C++_Programming/Practise_Programs/PP_030.cpp
//////////////////////////////////////////////////////////////////
//
//  File Name         : PP_030
//
//  Description       : Accepts an integer from the user and checks
//                      whether the number is even or odd.
//
//  Author            : Shubham Somanath Gadhe
//  Date              : 17/09/2026
//
//  Time Complexity   : O(1)
//  Space Complexity  : O(1)
//
//////////////////////////////////////////////////////////////////

#include<iostream>
using namespace std;

int main() 
{
    int number;

    cout << "Enter an integer: ";
    cin >> number;

    if (number % 2 == 0) 
    {
        cout << number << " is an even number" << endl;
    } 
    else 
    {
        cout << number << " is an odd number" << endl;
    }

    return 0;
}

//////////////////////////////////////////////////////////////////
//
//  Sample Input :
//      Enter an integer: 10
//
//  Sample Output :
//      10 is an even number
//
//////////////////////////////////////////////////////////////////